### ISSUE TITLE
Move virtual machines building process / hosting to Amazon S3 / EC2

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#509]
The virtual machine images are currently hosted at on a very unreliable server.

I'd be happy if someone could come up with a more cost-efficient solution than Amazon. But we need a reliable solution.
- The hosting is unreliable
- the automated build process stopped for some reason
- there is no VM for the 0.4 release
- there is no? error reporting

**Comment by nullusDefectus** [comment]
Will the VM image include the source code and the development environment as well? It will be extremely helpful if it did.

**Comment by StephanEwen** [comment]
I agree, it absolutely should. From what we have planned so far, the VM will include the following:
- An installation of Stratosphere (binaries)
- The source code
- Java 7 or 8 (OpenJDK)
- An IDE with the source code imported. I suggest to have the stratosphere engine projects initially closed (otherwise it may seem overwhelming if the entire source is showing/browsable)
- An initialized quickstart project for both Java and Scala
- A file on the desktop that explains what the VM provides 
- A script to pull the online documentation and the local test web server, to support fixing errors in the online documentation or providing additions
- A directory with small sample data for some common algorithms (text data, graph data, data for clustering algorithms)
- Links on the desktop to the homepage and the developer resources
- A python environment to try out the examples with visualization

We have someone working on a VM with the latest release build and on a solution that automatically builds a VM every night with the latest snapshot (nightly) build.

Please feel free to add to the list if you have an idea for other useful tools.
